dev/Tiles/entity.h

```diff
@@ -0,0 +1,66 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include "metatile.h"
+
+typedef struct Entity {
+    MetaTile* metatile[32];
+    unsigned char frameRate;
+    unsigned char numFrames;
```

Comment:
It's maybe a personal preference, but I kind of wince when I see loads of `unsigned char` types, and here's why:

- For general purpose programming, it's usually recommended to use the `int` type for integers. This will give you a signed integer of a size the compiler decides is most suited to the target architecture. When SDCC targets the Z80 it will mostly choose 16 bits for `int` IIRC according to the docs.
- If you are describing a data structure (which you are here) and you want to be very specific about the size, in that case I personally find the `char`, `short`, `long` etc. types far too abstract to be useful and would instead recommend using the very unambiguous types in `stdint.h`; i.e. `uint8_t` etc.

In fact, when I'm writing C code targeting embedded environments I tend to _only_ use the stdint types, unless I'm doing something like a for loop in which case I'll use the highly recommended simple `int`.

```suggestion
    uint8_t numFrames;
```